### PR TITLE
[JENKINS-73949] Extract inline event handlers from `LockableResourcesRootAction/tableResources/table.jelly`

### PR DESCRIPTION
**Pull request by yaroslavafenkin**
See [JENKINS-73949](https://issues.jenkins.io/browse/JENKINS-73949).

### Testing done

Create resources via the Global Configuration page. Then go to the lockable resources page. Click all the action buttons. Rendering of those is conditional, so  to see some of them there's a need to lock resources by running a job that acquires a lock on a resource.

For Note button it's straightforward. Click it and see textarea appear that lets add a note for a resource.

https://www.loom.com/share/8dcbe0b992164412af5def9829d2b44b

https://www.loom.com/share/828b59fbcf424475b7f45a0732fce2f6

### Proposed upgrade guidelines

N/A

### Localizations

N/A

### Submitter checklist

- [x] The Jira / Github issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - The changelog generator for plugins uses the **pull request title as the changelog entry**.
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during the upgrade.
- [ ] There is automated testing or an explanation that explains why this change has no tests.
- [ ] New public functions for internal use only are annotated with `@NoExternalUse`. In case it is used by non java code the `Used by {@code <panel>.jelly}` Javadocs are annotated.

- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease the future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
- [ ] Any localizations are transferred to *.properties files.
- [ ] Changes in the interface are documented also as [examples](src/doc/examples/readme.md).

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There is at least one (1) approval for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the **pull request title** and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically. See also [release-drafter-labels](https://github.com/jenkinsci/.github/blob/ce466227c534c42820a597cb8e9cac2f2334920a/.github/release-drafter.yml#L9-L50).
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] java code changes are tested by automated test.

### CODE REVIEW

File: src/test/java/org/jenkins/plugins/lockableresources/LockableResourceRootActionSEC1361Test.java [hunk 1]
```diff
@@ -80,8 +80,8 @@ private void checkXssWithResourceName(String resourceName) throws Exception {
 
         HtmlElement reserveButton = null;
         for (HtmlElement b : allButtons) {
-            String onClick = b.getAttribute("onClick");
-            if (onClick != null && onClick.contains("reserve")) {
+            String action = b.getAttribute("data-action");
+            if (action != null && action.contains("reserve")) {
```
Comment (yaroslavafenkin) [inline src/test/java/org/jenkins/plugins/lockableresources/LockableResourceRootActionSEC1361Test.java:84]: I'm not sure if this test makes enough sense to keep it around at this point. The code for which it was originally written has been changed to the point that it's barely recognizable.

Comment (basil) [reply src/test/java/org/jenkins/plugins/lockableresources/LockableResourceRootActionSEC1361Test.java:84]: A valid point; on the other hand, it does provide some automated coverage of the frontend, which is arguably valuable in and of itself…

**Review by basil — APPROVED** [review]
Thanks for the PR!

**Comment by yaroslavafenkin** [comment]
> Thanks for the PR!

Thanks for he formatting corrections. Also feel free to ask me to do those. Would happily do that when I'm around. Of course unless it's easier for you to fix stuff yourself.

**Pull request merged**